documentation/developers/mapping_instructions.md

```diff
@@ -0,0 +1,748 @@
+```
+src:    $.households[*].address 
+dst:    House.address
+```
+
+As you can see, the Destination instruction `House.address` looks like a JSONPath. However, a sample output of the above instruction would be:
+
+```json
+[
+    {"address" : "74 Whyioughta St."},
+    {"address" : "1600 Ipoke Ave"}
+]
+```
+`House` specifies the type of the object and tells Aether how to validate and where to save the data. It's not, strictly speaking, part of a valid JSONPath. This is why we don't use the `$` convention for Destination paths.
+
+Similarly, we can resolve the names of our Stooges.
+
+```
+src:    $.households[*].names[*]
+dst:    Person.name
+```
+
+This yields the expected:
+
+```json
+[
+    {"name" : "Larry"},
+    {"name" : "Curly"},
+    {"name" : "Moe"}
+]
+```
+
+---
+#### JSONPath extensions
+
+One annoying antipattern shows up in our sample input, the use of numbered keys like `number1` `number2` related keys in an object. JSONPath natively has no answer for this using the typical wildcard `*` operator. We've gone beyond the specification to enable the handling though. For example:
+
+```
+src:    $.households[*].number* 
+dst:    Person.name
+```
+We would expect this to yield:
+
+```json
+[
+    {"name" : "Larry", "number": 1},
+    {"name" : "Curly", "number": 2},
+    {"name" : "Moe", "number": 3}
+]
+```
+
+---
+#### JSONPath filtering and other functions
+
+As we use the extended parser in `jsonpath-ng` there are a number of functional improvements that you can leverage. A reference is available [here](https://github.com/h2non/jsonpath-ng#extensions).
+
+Filtering is an advanced feature and can change the order of the filtered field in the output entities. If you're creating more than one entity of a particular type, you must be sure that your outputs properly align to the source document as you're creating your mapping. The best way to do this is to:
+  - Apply non-overlapping filtering to all instances of any filtered type.
+  - Always apply filters in the same order.
+  - Apply the same filter for all properties of a given type.
+
+For more information about how the extractor interprets instructions, and why filtering order matters see the section titled [_Extractor Mechanism_](#user-content-extractor-mechanism).
+
+<a name="extractor-function"></a>
+## Extractor Functions
+
+##### UUID
+- *call:* `#!uuid`
+- *args:* `none`
+- *explanation:*
+
+    Generates a random UUID (version 4) which can be assigned as a string to a field. Beyond generating the ID, this is saved as part of the original submission document in a section titled `aether_extractor_enrichment` so that if extraction is performed a second time against the same dataset, the same IDs will be used. This expects that the order of the entities within a type will not change. See [_Extractor Mechanism_](#user-content-extractor-mechanism) for more information on instruction ordering.
```

Comment:
"This expects that the order of the entities within a type will not change" - I don't understand this?